perun-auditlogger/src/main/java/cz/metacentrum/perun/auditlogger/logger/impl/EventLoggerImpl.java

```diff
@@ -81,20 +81,21 @@ public void run() {
 			while (running) {
 
 				int sleepTime = 1000;
+				log.debug("Waiting for audit messages.");
 				//Waiting for new messages. If consumer failed in some internal case, waiting until it will be repaired (waiting time is increases by each attempt)
 				while(messages == null) {
 					try {
 						//IMPORTANT STEP1: Get new bulk of messages
 						messages = ((PerunBl)perun).getAuditMessagesManagerBl().pollConsumerMessages(perunSession, auditLoggerManager.getConsumerName(), lastProcessedIdNumber);
-						log.debug("Read {} new audit messages starting from {}", messages.size(), lastProcessedIdNumber);
+						if (messages.size() > 0) log.debug("Read {} new audit messages starting from {}", messages.size(), lastProcessedIdNumber);
 					} catch (InternalErrorException ex) {
 						log.error("Consumer failed due to {}. Sleeping for {} ms.", ex, sleepTime);
 						Thread.sleep(sleepTime);
 						sleepTime += sleepTime;
 					}
 
-					//If there are no messages, sleep for 1 sec and then try it again
-					if (messages == null) Thread.sleep(1000);
+					//If there are no messages, sleep for 3 sec and then try it again
+					if (messages == null) Thread.sleep(3000);
```

Comment:
`messages` are actually `null` only when this process is started. Method filling it with data returns empty list if there are no new messages. We should probably switch the initial `while` condition. Then we would keep this cycle going until new message arrives. Right now it still logs 3 messages each 5 seconds.
Also maybe use 5 seconds here, since outer cycle (`while (running)`) is doing the same.